code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,842 @@
+	desc = "Allows you to silently communicate with all other shadowlings and thralls."
+	panel = "Thrall Abilities"
+	charge_max = 50
+	human_req = 1
+	clothes_req = 0
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	action_icon_state = "commune"
+
+/obj/effect/proc_holder/spell/self/lesser_shadowling_hivemind/cast(mob/living/carbon/human/user)
+	if(!is_shadow_or_thrall(user))
+		to_chat(user, "<span class='warning'><b>As you attempt to commune with the others, an agonizing spike of pain drives itself into your head!</b></span>")
+		user.apply_damage(10, BRUTE, "head")
+		return
+	var/text = stripped_input(user, "What do you want to say your masters and fellow thralls?.", "Lesser Commune", "")
+	if(!text)
+		return
+	text = "<span class='shadowling'><b>\[Thrall\]</b><i> [user.real_name]</i>: [text]</span>"
+	for(var/mob/M in GLOB.mob_list)
+		if(is_shadow_or_thrall(M))
+			to_chat(M, text)
+		if(isobserver(M))
+			to_chat(M, "<a href='?src=\ref[M];follow=\ref[user]'>(F)</a> [text]")
+	log_say("[user.real_name]/[user.key] : [text]")
+
+
+// ASCENDANT ABILITIES BEYOND THIS POINT //
+// YES THEY'RE OP, BUT THEY'VE WON AT THE POINT WHERE THEY HAVE THIS, SO WHATEVER. //
+
+
+/obj/effect/proc_holder/spell/targeted/annihilate //Gibs someone instantly.
+	name = "Annihilate"
+	desc = "Gibs someone instantly."
+	panel = "Ascendant"
+	range = 7
+	charge_max = 0
+	clothes_req = 0
+	action_icon_state = "annihilate"
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+	sound = 'sound/magic/Staff_Chaos.ogg'
+
+/obj/effect/proc_holder/spell/targeted/annihilate/cast(list/targets,mob/living/simple_animal/ascendant_shadowling/user = usr)
+	if(user.incorporeal_move)
+		to_chat(user, "<span class='warning'>You are not in the same plane of existence. Unphase first.</span>")
+		revert_cast()
+		return
+	for(var/mob/living/boom in targets)
+		if(is_shadow(boom)) //Used to not work on thralls. Now it does so you can PUNISH THEM LIKE THE WRATHFUL GOD YOU ARE.
+			to_chat(user, "<span class='warning'>Making an ally explode seems unwise.<span>")
+			revert_cast()
+			return
+
+		if (!boom.is_holding(/obj/item/storage/backpack/holding)) //so people actually have a chance to kill ascended slings without being insta-sploded
+			user.visible_message("<span class='warning'>[user]'s markings flare as they gesture at [boom]!</span>", \
+								"<span class='shadowling'>You direct a lance of telekinetic energy into [boom].</span>")
+			if(iscarbon(boom))
+				playsound(boom, 'sound/magic/Disintegrate.ogg', 100, 1)
+			boom.visible_message("<span class='userdanger'>[boom] explodes!</span>")
+			boom.gib()
+		else
+			to_chat(user, "<span class='warning'>The telekinetic energy is absorbed by the bluespace portal in [boom]'s hand!<span>")
+			to_chat(boom, "<span class='userdanger'>You feel a slight recoil from the bag of holding!<span>")
+
+
+/obj/effect/proc_holder/spell/targeted/hypnosis //Enthralls someone instantly. Nonlethal alternative to Annihilate
+	name = "Hypnosis"
```

Comment:
this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.